### PR TITLE
Add checking to write and confirm order

### PR DESCRIPTION
**Pull request by TimLai125**

### CODE REVIEW

File: purchase_ext_sst/models/purchase_order.py [hunk 1]
```diff
@@ -103,27 +103,29 @@ def open_record(self):
...
-                partner_id = self.create_partner(val_phone or
-                                                 val_phone_search,
-                                                 val_mobile).id
-            if not self.is_default_partner(partner_id):
-                partner = self.env['res.partner'].browse(partner_id)
-                self.update_partner(partner, val_phone, val_mobile,
-                                    val_tent_name)
-        vals['partner_id'] = partner_id
-        vals['phone_update'] = vals['mobile_update'] = False
+        if 'phone_search' in vals or 'phone_update' in vals or \
```
Comment (TimLai125) [inline purchase_ext_sst/models/purchase_order.py]: @yostashiro Do you think we should add `'partner_id' in vals` in here as well? I cannot imagine if there is cases that need this statement.

Comment (yostashiro) [reply purchase_ext_sst/models/purchase_order.py]: @TimLai125 I'm a bit lost.  What issue are you trying to solve here?

Comment (TimLai125) [reply purchase_ext_sst/models/purchase_order.py]: @yostashiro Here is used to solve the purchase order cancel issue. When multiple orders is selected in the tree view and use the "Cancel Purchase Orders" button, the `partner_id` of all `purchase.order` will be updated.

**Comment by TimLai125** [comment]
Take over by #101

**Pull request closed without merging**